### ISSUE TITLE
[BUG] Error waiting for API server

### ISSUE THREAD

**Issue by MAHDTech** [code-ready/crc#2484]
### TLDR

crc start fails with `error waiting for apiserver`. If left for a few minutes without user intenvetion, crc reports healthy and operates successfully.

### General information

  * OS: Fedora Server 34
  * Hypervisor: KVM
  * Did you run `crc setup` before starting it (Yes/No)? Yes
  * Running CRC on: Baremetal-Server

## CRC version

```bash
CodeReady Containers version: 1.28.0+08de64bd
OpenShift version: 4.7.13 (embedded in executable)
```
  
## CRC status

```bash
DEBU CodeReady Containers version: 1.28.0+08de64bd 
DEBU OpenShift version: 4.7.13 (embedded in executable) 
DEBU Running 'crc status'                         
DEBU Checking file: /home/linadmin/.crc/machines/crc/.crc-exist 
DEBU Checking file: /home/linadmin/.crc/machines/crc/.crc-exist 
DEBU Found binary path at /home/linadmin/.crc/bin/crc-driver-libvirt 
DEBU Launching plugin server for driver libvirt   
DEBU Plugin server listening at address 127.0.0.1:33725 
DEBU () Calling .GetVersion                       
DEBU Using API Version 1                          
DEBU () Calling .SetConfigRaw                     
DEBU () Calling .GetMachineName                   
DEBU (crc) Calling .GetState                      
DEBU (crc) DBG | time="2021-06-28T13:42:41+09:30" level=debug msg="Getting current state..." 
DEBU (crc) DBG | time="2021-06-28T13:42:41+09:30" level=debug msg="Fetching VM..." 
DEBU (crc) Calling .GetBundleName                 
DEBU (crc) Calling .GetIP                         
DEBU (crc) DBG | time="2021-06-28T13:42:41+09:30" level=debug msg="GetIP called for crc" 
DEBU (crc) DBG | time="2021-06-28T13:42:41+09:30" level=debug msg="Getting current state..." 
DEBU (crc) DBG | time="2021-06-28T13:42:41+09:30" level=debug msg="IP address: 192.168.130.11" 
DEBU Running SSH command: df -B1 --output=size,used,target /sysroot | tail -1 
DEBU Using ssh private keys: [/home/linadmin/.crc/machines/crc/id_ecdsa /home/linadmin/.crc/cache/crc_libvirt_4.7.13/id_ecdsa_crc] 
DEBU SSH command results: err: <nil>, output: 106825756672 12537110528 /sysroot 
DEBU cannot get OpenShift status: Get "https://api.crc.testing:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 192.168.130.11:6443: connect: connection refused 
DEBU Making call to close driver server           
DEBU (crc) Calling .Close                         
DEBU Successfully made call to close driver server 
DEBU Making call to close connection to plugin binary 
DEBU (crc) DBG | time="2021-06-28T13:42:42+09:30" level=debug msg="Closing plugin on server side" 
CRC VM:          Running
OpenShift:       Unreachable (v4.7.13)
Disk Usage:      12.54GB of 106.8GB (Inside the CRC VM)
Cache Usage:     13.58GB
Cache Directory: /home/linadmin/.crc/cache
```

## CRC config

```bash
- consent-telemetry                     : yes
- cpus                                  : 6
- disk-size                             : 100
- memory                                : 15000
- pull-secret-file                      : /home/linadmin/.crc/pull-secret.json
```

## Host Operating System

```bash
NAME=Fedora
VERSION="34 (Server Edition)"
ID=fedora
VERSION_ID=34
VERSION_CODENAME=""
PLATFORM_ID="platform:f34"
PRETTY_NAME="Fedora 34 (Server Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:34"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/34/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=34
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=34
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Server Edition"
VARIANT_ID=server
```

### Steps to reproduce

  1. `crc delete -f`
  2. `crc cleanup`
  3. `crc setup`
  4. `crc start --log-level debug`

### Expected

CRC setup completed without error.

### Actual

CRC exits with `Error waiting for API server`.

```bash
crc start --log-level debug
<snip>
Error waiting for apiserver: Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 1\n (x7)
Temporary error: ssh command error:
command : timeout 5s oc get nodes --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 124\n (x38)
```

After a few minutes, CRC starts successfully, as shown by running status again.

```bash
DEBU CodeReady Containers version: 1.28.0+08de64bd 
DEBU OpenShift version: 4.7.13 (embedded in executable) 
DEBU Running 'crc status'                         
DEBU Checking file: /home/linadmin/.crc/machines/crc/.crc-exist 
DEBU Checking file: /home/linadmin/.crc/machines/crc/.crc-exist 
DEBU Found binary path at /home/linadmin/.crc/bin/crc-driver-libvirt 
DEBU Launching plugin server for driver libvirt   
DEBU Plugin server listening at address 127.0.0.1:38397 
DEBU () Calling .GetVersion                       
DEBU Using API Version 1                          
DEBU () Calling .SetConfigRaw                     
DEBU () Calling .GetMachineName                   
DEBU (crc) Calling .GetState                      
DEBU (crc) DBG | time="2021-06-28T13:52:36+09:30" level=debug msg="Getting current state..." 
DEBU (crc) DBG | time="2021-06-28T13:52:36+09:30" level=debug msg="Fetching VM..." 
DEBU (crc) Calling .GetBundleName                 
DEBU (crc) Calling .GetIP                         
DEBU (crc) DBG | time="2021-06-28T13:52:36+09:30" level=debug msg="GetIP called for crc" 
DEBU (crc) DBG | time="2021-06-28T13:52:36+09:30" level=debug msg="Getting current state..." 
DEBU (crc) DBG | time="2021-06-28T13:52:36+09:30" level=debug msg="IP address: 192.168.130.11" 
DEBU Running SSH command: df -B1 --output=size,used,target /sysroot | tail -1 
DEBU Using ssh private keys: [/home/linadmin/.crc/machines/crc/id_ecdsa /home/linadmin/.crc/cache/crc_libvirt_4.7.13/id_ecdsa_crc] 
DEBU SSH command results: err: <nil>, output: 106825756672 12652421120 /sysroot 
DEBU Making call to close driver server           
DEBU (crc) Calling .Close                         
DEBU Successfully made call to close driver server 
DEBU Making call to close connection to plugin binary 
DEBU (crc) DBG | time="2021-06-28T13:52:37+09:30" level=debug msg="Closing plugin on server side" 
CRC VM:          Running
OpenShift:       Running (v4.7.13)
Disk Usage:      12.65GB of 106.8GB (Inside the CRC VM)
Cache Usage:     13.58GB
Cache Directory: /home/linadmin/.crc/cache
```

### Logs

Before gather the logs try following if that fix your issue

```bash
$ crc delete -f
$ crc cleanup
$ crc setup
$ crc start --log-level debug
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

- [crc startup debug log](https://gist.github.com/MAHDTech/9471c217a2cc46a1cc01b6d52de06c42)

Additional logs gathered from within the CRC VM.

- [kubelet logs](https://gist.github.com/MAHDTech/a8e327766413c54781b9452b2d15690c)

- [crictl ps](https://gist.github.com/MAHDTech/b8818578538bd2ba665a4c07b6eb7a9c)

- [kube-apiserver logs](https://gist.github.com/MAHDTech/b9d5d127fc88fd946465a086441bb552)

```bash
KUBECONFIG=/opt/kubeconfig kubectl get nodes

NAME                 STATUS   ROLES           AGE   VERSION
crc-pkjt4-master-0   Ready    master,worker   17d   v1.20.0+df9c838
```

### Related

The same error message is reported in the following issues so likely related however with different os/version combinations.

- Issue #2160
- Issue #2384
